Shared/Samples/Show device location with NMEA data sources/ShowDeviceLocationWithNMEADataSourcesView.Model.swift

```diff
@@ -0,0 +1,296 @@
+// Copyright 2022 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Combine
+import ExternalAccessory
+
+extension ShowDeviceLocationWithNMEADataSourcesView {
+    /// The model used to store the geo model and other expensive objects used in this view.
+    class Model: ObservableObject {
+        // MARK: Properties
+        
+        /// A map with a navigation basemap.
+        let map = Map(basemapStyle: .arcGISNavigation)
+        
+        /// The current autopan mode.
+        var autoPanMode: LocationDisplay.AutoPanMode {
+            get {
+                locationDisplay.autoPanMode
+            }
+            set {
+                locationDisplay.autoPanMode = newValue
+            }
+        }
+        
+        /// A Boolean value specifying if the "recenter" button should be disabled.
+        @Published private(set) var isRecenterButtonDisabled = true
+        
+        /// A Boolean value specifying if the "reset" button should be disabled.
+        @Published private(set) var isResetButtonDisabled = true
+        
+        /// A Boolean value specifying if the "source" button should be disabled.
+        @Published private(set) var isSourceMenuDisabled = false
+        
+        /// A string containing GPS accuracy.
+        @Published private(set) var accuracyStatus = "Accuracy info will be shown here."
+        
+        /// A string containing satellite information.
+        @Published private(set) var satelliteStatus = "Satellites info will be shown here."
+        
+        /// The location display used in the map view.
+        let locationDisplay: LocationDisplay = {
+            let locationDisplay = LocationDisplay()
+            locationDisplay.autoPanMode = .recenter
+            return locationDisplay
+        }()
+        
+        /// An NMEA location data source, to parse NMEA data.
+        private var nmeaLocationDataSource: NMEALocationDataSource!
+        
+        /// A mock data source to read NMEA sentences from a local file, and generate
+        /// mock NMEA data every fixed amount of time.
+        private let mockNMEADataSource = SimulatedNMEASentenceFeed(
```

Comment:
if the type is called `SimulatedNMEASentenceFeed` then the variable should be called mockNMEAFeed